python/wakapi_anyide/runner.py

```diff
@@ -84,7 +84,7 @@ async def heartbeat_task(env: Environment, queue: Queue[Event], watchers: Sequen
             "category": "coding",
             "time": event.time,
             "project": env.project.project.name,
-            "language": Path(event.filename).suffix.replace('.', '') or Path(event.filename).name,
+            **({"language": languageProcessor(env, event.filename)} if languageProcessor(env, event.filename) is not None else {}), # dict comprehension, only include language if it's custom defined, otherwise let Wakatime determine,
```

Comment:
Hackatime does not autodetect from the filename, it simply says "Unknown". Return the file extension if the language is not known.